asdf/core/_converters/complex.py

```diff
@@ -0,0 +1,22 @@
+import re
+
+from asdf.extension import Converter
+
+_REPLACEMENTS = {
+    re.compile("i(?!nf)"): "j",
+    re.compile("I(?!NF)"): "J",
+}
+
+
+class ComplexConverter(Converter):
+    tags = ["tag:stsci.edu:asdf/core/complex-1.0.0"]
+    types = ["builtins.complex", "numpy.complex64", "numpy.complex128", "numpy.complex256"]
```

Comment:
The old code dynamically inspected numpy to get all complex types. I think the explicit list introduced here is more readable. However, on my system I do not have a `complex256` and I do have a `cdoublelong` (returned from `util.iter_subclasses` like in the old code) which appears to be 128 bit but fails isinstance checks against complex128 and complex64.

Also, it looks like this and the previous code did not roundtrip numpy objects. On my system the builtin complex seems capable of representing the largest (complex128) values that numpy supports so this might not be an issues but I wonder if large complex256 will become infs. Does your system support complex256 and can you test this?

Reply:
Given the varied support across machines I'm thinking matching the old type (using `iter_subclasses`) is preferable so that we don't miss types.

Your point about tightening up the standard is a good one and deserves a separate issue. Do you think it makes sense to open one in ASDF or ASDF-standard?

Failure to roundtrip that preserves precision seems fine to me, failures to roundtrip that lose precision (and/or convert numbers to inf/nan) seem unacceptable. I'm all ears for ideas on how to best handle this and must admit that I don't fully grasp all the places where precision loss can occur (on write/conversion to new type, then to yaml, on read/conversion from yaml then to new type, ect).